neon_messagebus_mq_connector/controller.py

```diff
@@ -134,89 +149,144 @@ def handle_neon_profile_update(self, message: Message):
+                                                             'gender': message.data.get('gender', 'female'),
+                                                             'created_on': int(time.time())}
 
     def handle_user_message(self,
                             channel: pika.channel.Channel,
                             method: pika.spec.Basic.Return,
                             properties: pika.spec.BasicProperties,
                             body: bytes):
         """
-        Handles requests from MQ to Neon Chat API received on queue
-        "neon_chat_api_request"
+            Transfers requests from MQ API to Neon Message Bus API
 
-        :param channel: MQ channel object (pika.channel.Channel)
-        :param method: MQ return method (pika.spec.Basic.Return)
-        :param properties: MQ properties (pika.spec.BasicProperties)
-        :param body: request body (bytes)
+            :param channel: MQ channel object (pika.channel.Channel)
+            :param method: MQ return method (pika.spec.Basic.Return)
+            :param properties: MQ properties (pika.spec.BasicProperties)
+            :param body: request body (bytes)
 
         """
         if body and isinstance(body, bytes):
             dict_data = b64_to_dict(body)
             LOG.info(f'Received user message: {dict_data}')
-            dict_data["context"].setdefault("mq", dict())
-            if "routing_key" in dict_data:
-                dict_data["context"]["mq"]["routing_key"] = \
-                    dict_data.pop("routing_key")
-            if "message_id" in dict_data:
-                dict_data["context"]["mq"]["message_id"] = \
-                    dict_data.pop("message_id")
-            check_error, dict_data = self.validate_request(dict_data)
-            if check_error is not None:
-                LOG.error(check_error)
+            dict_data["context"].setdefault("mq", dict(routing_key=dict_data.pop('routing_key', ''),
+                                                       message_id=dict_data.pop('message_id', '')))
+
+            validation_error, dict_data = self.validate_request(dict_data)
+            if validation_error:
                 response = Message(msg_type="klat.error",
-                                   data=dict(error=str(check_error),
+                                   data=dict(error=validation_error,
                                              message=dict_data))
-                self.handle_neon_message(response, "neon_chat_api_error")
+                response.context.setdefault('klat', {})['routing_key'] = 'neon_chat_api_error'
+                self.handle_neon_message(response)
             else:
+                dict_data["context"].setdefault('ident', f"{dict_data['msg_type']}.response")
                 message = Message(**dict_data)
-                if message.msg_type in ("neon.get_stt", "neon.get_tts",
-                                        "neon.audio_input"):
+                self.bus.emit(message)
```

Comment:
emit should happen after call to `push_awaiting_message` to avoid a potential race condition (in case response is received before the message is put in the internal list here